### DISCUSSION TITLE
Clarification Needed on TypeScript as a Peer Dependency

### DISCUSSION THREAD

**Discussion by alex-vladut** [0no-co/gql.tada#245] [Q&A] [answered]
Hello,

I'm currently integrating the `gql.tada` package into our project and I've noticed that `typescript` is listed as both a peer dependency and a dev dependency in your package. Given the recent behaviour of npm (since version 7), `npm install` automatically includes peer dependencies, which means `typescript` gets installed in environments where it might not be necessary, such as production. We are deploying the bundle as an AWS Lambda function.

**Questions:**
1. **Why is TypeScript listed as a peer dependency?** Could anyone provide insight into the decision behind this, especially since it's also included as a dev dependency?
2. **Impact on Production Environments:** Considering that `typescript` is typically not required at runtime, is there a particular reason it needs to be installed by end-users of `gql.tada`? If its usage is confined to development, could it be restricted to dev dependencies to optimise dependency management in production deployments?

Our goal here is to streamline production deployments by minimising unnecessary installations and as a result reducing the bundle size.

Thank you 🙏

**Comment by kitten** [comment, marked as answer]
You can take a look at the package structure and their dependencies:
- https://github.com/0no-co/gql.tada/blob/main/package.json
- https://github.com/0no-co/gql.tada/blob/main/packages/internal/package.json
- https://github.com/0no-co/gql.tada/blob/main/packages/cli-utils/package.json

As it stands, `gql.tada` has a single runtime dependency on `@0no-co/graphql.web`.

However, it installs `@gql.tada/cli-utils` as a dependency, since we currently haven't added a mechanism to install this when the CLI is run and didn't want to spend time on this.

The CLI is split into reusable and non-reusable parts for code-sharing with `@0no-co/graphqlsp`. Reusable parts are shared in `@gql.tada/internal`.
The CLI relies on TypeScript to execute most of its pre-compilation tasks.

Re. `peerDependencies` and `devDependencies`. The latter isn't relevant for you since it's for development in our environment and is to fulfill our own peer on `typescript`. It's expected that `typescript `is installed under `devDependencies` in your own project to fulfil this requirement, hence a `--prod` install won't include it.
However, currently, `ts-morph` and other dependencies will still be installed in a `--prod` install.

If you're specifically catering your output for AWS Lambdas (et al; and don't want to bundle dependencies) then I can currently only recommend something like [`trace-deps`](https://www.npmjs.com/package/trace-deps) to guarantee that dependencies not used during runtime aren't installed

**Reply by alex-vladut** [reply]
Thank you very much for your quick reply!
